packages/libs/eda/src/lib/core/utils/visualization.ts

```diff
@@ -259,11 +273,19 @@ export function assertValidInputVariables(
     );
   });
   if (invalidInputs.length) {
-    throw new Error(
-      `The following variables are invalid and must be changed: ${invalidInputs
-        .map((input) => input.label)
-        .join(', ')}`
-    );
+    if (dataElementDependencyOrder != null) {
+      throw new Error(
+        `The variables are no longer valid due to entity constraints. Try disabling or changing the ${
+          invalidInputs[invalidInputs.length - 1].label
+        } variable.`
+      );
+    } else {
```

Comment:
This logic seems flawed. For example, this condition will be met for a variable of an acceptable entity, but with too many values (such as when a filter is removed from a categorical variable with more than 8 values).

I think we will have to extract some logic from `disabledVariablesForInput` to be this specific.

Reply:
> this condition will be met for a variable of an acceptable entity, but with too many values (such as when a filter is removed from a categorical variable with more than 8 values).

Oh, I guess this isn't an issue in the SAM, since we get the top 7+other values.

Reply:
I tried to handle "non-entity constraints" first by calling the function from itself without the `dataElementDependencyOrder` - so the user *should* get a more helpful message if one of those constraints breaks first?

Reply:
Oh, I see! I should have read the comments :flushed:.